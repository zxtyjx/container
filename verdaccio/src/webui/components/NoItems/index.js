import React from 'react';
import PropTypes from 'prop-types';

import classes from './noItems.scss';

const NoItems = (props) => {
    return (
      <div className={classes.noItems}>
        <h2>{props.text}</h2>
      </div>
    );
};

NoItems.propTypes = {
  text: PropTypes.string.isRequired
};

export default NoItems;
