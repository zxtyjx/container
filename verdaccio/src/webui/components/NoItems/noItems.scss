.noItems {
  margin: 5em 0;
}
