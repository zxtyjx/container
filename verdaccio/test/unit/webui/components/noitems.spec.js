/**
 * NoItems component
 */

import React from 'react';
import { shallow } from 'enzyme';
import NoItems from '../../../../src/webui/components/NoItems/index';

console.error = jest.fn();

describe('<NoItem /> component', () => {
  it('should give error for the required fields', () => {
    shallow(<NoItems />);
    expect(console.error).toBeCalled();
  });

  it('should set html from props', () => {
    const props = {
      text: 'This is a test string'
    };
    const wrapper = shallow(<NoItems {...props} />);
    expect(wrapper.find('h2').text()).toEqual('This is a test string');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
