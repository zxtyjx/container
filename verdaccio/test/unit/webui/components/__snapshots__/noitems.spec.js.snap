// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoItem /> component should set html from props 1`] = `"<div class=\\"noItems\\"><h2>This is a test string</h2></div>"`;
